### DISCUSSION TITLE
Vulnerability due to Misconfiguration in the Microsoft IDP

### DISCUSSION THREAD

**Discussion by cht-wtag** [keycloak/keycloak#21209] [Q&A]
Recently, I came across this post [nOAuth: How Microsoft OAuth Misconfiguration Can Lead to Full Account Takeover](https://www.descope.com/blog/post/noauth) about the vulnerability of using Microsoft as a social login IDP(identity provider).

As per the [nOAuth: How Microsoft OAuth Misconfiguration Can Lead to Full Account Takeover](https://www.descope.com/blog/post/noauth) the attack vectors are -

- The attacker access Azure AD account as admin.
- The attacker changes the “Email” attribute to the victim’s email address.
- The attacker uses “Log in with Microsoft” on a website/app that’s vulnerable if uses the email address as a unique identifier.
- If the app merges user accounts without validation, the attacker now has full control over the victim’s account, even if the victim doesn’t have a Microsoft account.

I searched how the keycloak integrates the identity providers' [Server Administration Guide](https://www.keycloak.org/docs/latest/server_admin/#_identity_broker), but could not find any specific mention of the [insecure validating claims](https://learn.microsoft.com/en-us/azure/active-directory/develop/claims-validation#validate-the-subject)(`email`,` preferred_username`, `upn`, and `unique_name`) to store or determine whether the user in an access token should have access to data.

I found we have the option for enabling trust email for the email of the [identity providers](https://www.keycloak.org/docs/latest/server_admin/#_identity_broker). 
![image-20230622-113740](https://github.com/keycloak/keycloak/assets/57312839/28edecc1-15dc-4643-90f4-0ead0ee8595b)

Does enabling this can create any vulnerabilities? Also, we are using keyclaok **18.0.0**, does this version expose us for this particular vulnerability?

**Comment by abstractj** [comment]
@cht-wtag I want to highlight, how it's crucial to follow our responsible disclosure guidelines for security issues to protect other users. Please read: https://www.keycloak.org/security.html.
 
Openly sharing vulnerabilities can put others at risk. Added to that, please ensure you're running the latest version of Keycloak, as it includes the most recent security bug fixes. The issue, you mentioned, can be a misconfiguration issue when people enable automatic account linking without validation. That can be the cause of the issue reported here, and it goes against the security best practices recommended in our docs. Please see: https://github.com/keycloak/keycloak/issues/21155.
